src/crypto_backend.rs

```diff
@@ -18,13 +23,44 @@ pub trait LocalKeyPair {
     /// Export the public key component in the
     /// binary uncompressed point representation.
     fn pub_as_raw(&self) -> Result<Vec<u8>>;
+    /// Import a keypair from its raw components.
+    fn from_raw_components(components: &EcKeyComponents) -> Result<Self>
+    where
+        Self: Sized;
+    /// Export the raw components of the keypair.
+    fn raw_components(&self) -> Result<(EcKeyComponents)>;
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[cfg(feature = "serializable-keys")]
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct EcKeyComponents {
+    // The `d` value of the EC Key.
+    private_key: Vec<u8>,
+    // The uncompressed x,y-representation of the public component of the EC Key.
+    public_key: Vec<u8>,
```

Comment:
I kinda wonder if we might want to break these apart rather than the joined, zero pad value.